File: starter_gemini.py
"""
PromptWars - Gemini API Starter Client (Python)
Usage:
    pip install google-genai
    python starter_gemini.py
"""
import os
from google import genai
from google.genai import types

def get_ai_response(prompt: str, system_instruction: str = "You are an expert AI assistant for PromptWars.") -> str:
    api_key = os.environ.get("GEMINI_API_KEY")
    if not api_key:
        raise ValueError("Please set the GEMINI_API_KEY environment variable.")
        
    client = genai.Client(api_key=api_key)
    
    response = client.models.generate_content(
        model="gemini-2.5-flash",
        contents=prompt,
        config=types.GenerateContentConfig(
            system_instruction=system_instruction,
            temperature=0.7,
        )
    )
    return response.text

if __name__ == "__main__":
    test_prompt = "Give a 1-line winning pitch for a PromptWars hackathon AI solution."
    print("Testing Gemini Client...")
    try:
        res = get_ai_response(test_prompt)
        print("Response:\n", res)
    except Exception as e:
        print(f"Error: {e}")

File: starter_gemini.js
/**
 * PromptWars - Gemini API Starter Client (JavaScript / Node.js)
 * Usage:
 *   npm install @google/genai dotenv
 *   node starter_gemini.js
 */
import { GoogleGenAI } from "@google/genai";
import "dotenv/config";

const apiKey = process.env.GEMINI_API_KEY;
if (!apiKey) {
  console.error("Missing GEMINI_API_KEY in environment variables.");
}

const ai = new GoogleGenAI({ apiKey });

export async function askGemini(prompt, systemInstruction = "You are an expert AI assistant for PromptWars.") {
  const response = await ai.models.generateContent({
    model: "gemini-2.5-flash",
    contents: prompt,
    config: {
      systemInstruction: systemInstruction,
      temperature: 0.7,
    },
  });

  return response.text;
}

// Quick Test
if (process.argv[1] === import.meta.filename) {
  console.log("Testing JS Gemini Client...");
  askGemini("Give a 1-line punchy slogan for an AI Hackathon project.")
    .then((res) => console.log("Response:\n", res))
    .catch((err) => console.error("Error:", err.message));
}
